app/components/phone_input_component.rb

```diff
@@ -1,29 +1,37 @@
 class PhoneInputComponent < BaseComponent
-  attr_reader :form, :required
+  attr_reader :form, :required, :allowed_countries
   alias_method :f, :form
 
-  def initialize(form:, required: false)
+  def initialize(form:, allowed_countries: nil, required: false)
+    @allowed_countries = allowed_countries
     @form = form
     @required = required
   end
 
   def supported_country_codes
-    PhoneNumberCapabilities::INTERNATIONAL_CODES.keys
+    allowed_countries || PhoneNumberCapabilities::INTERNATIONAL_CODES.keys
```

Comment:
Maybe validate against actual codes:

```suggestion
    codes = PhoneNumberCapabilities::INTERNATIONAL_CODES.keys
    codes = codes & allowed_countries if allowed_countries
    codes
```